codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -307,33 +485,84 @@ public void onGoAwayReceived(final int lastStreamId, long errorCode, ByteBuf deb
         }
     }
 
-    private static final class InternalHttp2ConnectionHandler extends Http2ConnectionHandler {
+    private final class InternalHttp2ConnectionHandler extends Http2ConnectionHandler {
         InternalHttp2ConnectionHandler(Http2ConnectionDecoder decoder, Http2ConnectionEncoder encoder,
                                        Http2Settings initialSettings) {
             super(decoder, encoder, initialSettings);
         }
 
         @Override
+        protected void onConnectionError(ChannelHandlerContext ctx, Throwable cause, Http2Exception http2Ex) {
+            ctx.fireExceptionCaught(cause);
+        }
+
+        /**
+         * Exceptions for unknown streams, that is streams that have no {@link Http2Stream2} object attached
+         * are simply logged and replied to by sending a RST_STREAM frame. There is not much value in propagating such
+         * exceptions through the pipeline, as a user will not have any additional information / state about this
+         * stream and thus can't do any meaningful error handling.
+         */
+        @Override
         protected void onStreamError(ChannelHandlerContext ctx, Throwable cause,
-                                     Http2Exception.StreamException http2Ex) {
-            try {
-                Http2Stream stream = connection().stream(http2Ex.streamId());
-                if (stream == null) {
-                    return;
-                }
-                ctx.fireExceptionCaught(http2Ex);
-            } finally {
-                super.onStreamError(ctx, cause, http2Ex);
+                                     Http2Exception.StreamException streamException) {
+            int streamId = streamException.streamId();
+            Http2Stream connectionStream = super.connection().stream(streamId);
+            if (connectionStream == null) {
+                LOG.warn("Stream exception thrown for unkown stream.", cause);
+                // Write a RST_STREAM
+                super.onStreamError(ctx, cause, streamException);
+                return;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Prefix this call to "connection" with "super.". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2388)